src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Commands/AddCohort/AddCohortHandler.cs

```diff
@@ -64,4 +75,25 @@ public async Task<AddCohortResult> Handle(AddCohortCommand command, Cancellation
 
         return response;
     }
+
+    private async Task CreateAutoReservationAndAdd(AddCohortCommand command, Cohort cohort, CancellationToken cancellationToken)
+    {
+        Guid.TryParse(command.UserInfo.UserId, out Guid userId);
+
+        var accountLegalEntity = await _dbContext.Value.AccountLegalEntities.Where(x => x.Id == command.AccountLegalEntityId).FirstOrDefaultAsync();
+        var request = new CreateAutoReservationRequest
+        {
+            AccountId = command.AccountId,
+            AccountLegalEntityId = command.AccountLegalEntityId,
+            CourseId = command.CourseCode,
+            AccountLegalEntityName = accountLegalEntity?.Name,
+            ProviderId = Convert.ToUInt32(command.ProviderId),
+            StartDate = command.StartDate.Value,
+            UserId = userId
+        };
+
+        var response = await _reservationsClient.CreateAutoReservation(request, cancellationToken);
```

Comment:
I didn't like doing it this way, but we already do this in other places, CoE/CoP and it's slightly cleaner than having to return to the Outer API by throwing some exception (as a SQL transaction is already in place), and if we don't throw the exception the record may be saved implicitly. 

The outer API would then have to capture this exception (handle it differently to domain exceptions) and then create the reservation (I agree this means we are creating a reservation in the correct place), add that to the new cohort/apprenticeship request we just submitted and resubmit it.

An alternative is to create the draft apprenticeship with the reservation (which is possible) and then create the reservation and add that to this apprenticeship, but that risks having a draft apprenticeship without a reservation is something goes wrong.

 Another option is to create the reservation first before we add the apprenticeship, but because all validations are done   during saving of the data, this leads to lots of other issues.